crates/holochain_cascade/src/lib.rs

```diff
@@ -1084,3 +1116,76 @@ where
         }
     }
 }
+
+impl MockCascade {
+    /// Construct a mock which acts as if the given records were part of local storage
+    pub fn with_records(records: Vec<Record>) -> Self {
+        let mut cascade = Self::default();
+
+        let map: HashMap<AnyDhtHash, Record> = records
+            .into_iter()
+            .flat_map(|r| {
+                let mut items = vec![(r.action_address().clone().into(), r.clone())];
+                if let Some(eh) = r.action().entry_hash() {
+                    items.push((eh.clone().into(), r))
+                }
+                items
+            })
+            .collect();
+
+        let map0 = ShareOpen::new(map);
+
+        let map = map0.clone();
+        cascade.expect_retrieve().returning(move |hash, _| {
+            box_fut_plain(Ok(map.share_ref(|m| {
```

Comment:
"fut"?

Reply:
"future", this is meant to reduce boilerplate so I chose a short name